lms/services/blackboard_api/service.py

```diff
@@ -59,6 +65,24 @@ def list_files(self, course_id):
             schema=BlackboardListFilesSchema,
         ).validated_data
 
+    def public_url(self, course_id, document_url):
+        # Get just the file_id part out of the
+        # blackboard://content-resource/file_id/ URL.
+        file_id = re.search(
```

Comment:
Should we compile the regexp, move it out of the method and use it here? Some perf benefit but less "code locality". Not strong preference from my side.

Reply:
Done